lib/faye/redis.rb

```diff
@@ -180,15 +180,20 @@ def redis
         @subscriber.on(:disconnected) do
           @server.info "Faye::Redis: redis pubsub connection disconnected"
         end
-       @subscriber.on(:reconnected) do
+        @subscriber.on(:reconnected) do
           @server.info "Faye::Redis: redis pubsub connection reconnected"
         end
         @subscriber.on(:reconnect_failed) do |count|
-          @server.info "Faye::Redis: redis pubsub connection reconnect failed (#{count}/4)"
+          @server.info "Faye::Redis: redis pubsub connection reconnect failed #{count} time(s)"
+          begin
+            fn = @options[:reconnect_failed] || ->(_, _) {}
+            fn.call(count, 'pubsub')
```

Comment:
One usage for this hook is to send notifications to an error monitoring product such as Bugsnag.